presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -408,6 +423,10 @@ protected Boolean visitDereferenceExpression(DereferenceExpression node, Void co
 
         private Boolean isField(QualifiedName qualifiedName)
         {
+            if (lambdaVariableNames.contains(qualifiedName.toString())) {
```

Comment:
At this point we lost information whether it was `a` or `t.a`.. That would be better handled in the `visitIdentifier` method: if `lambdaVariableNames.contains(..)`. then don't call `isField`.